Add support for bounded enumerables and stateless Procs in inclusion validation (#1040)
Shopify/maintenance_tasks: README.md

inputs, it’s important to check that they conform to the format your Task
expects, and to sanitize any inputs if necessary.

#### Validating Task Parameters

Task attributes can be validated using Active Model Validations. Attributes are
validated before a Task is enqueued.

If an attribute uses an inclusion validator with a supported `in:` option, the
set of values will be used to populate a dropdown in the user interface. The
following types are supported:

* Arrays

> * Procs and lambdas that optionally accept the instance instance, and return an Array.
> * Callable objects that receive one argument, the task instance, and return an Array.
> * Methods that return an Array, called on the task instance.

Reviewer: ```suggestion
* Procs and lambdas that optionally accept the Task instance, and return an Array.
* Callable objects that receive one argument, the Task instance, and return an Array.
* Methods that return an Array, called on the Task instance.
```

inputs, it’s important to check that they conform to the format your Task
expects, and to sanitize any inputs if necessary.

#### Validating Task Parameters

> Task attributes can be validated using Active Model Validations. Attributes
> are validated when a Run is created, and validation errors are recorded on the
> Run.
> 
> If an attribute uses an inclusion validator with a supported enumerable type,
> the set of values will be used to populate a dropdown in the user interface. The
> following types are supported:

Reviewer: "recorded" is a bit misleading, we don't persist the errors. Also I don't think we should mention the Run.
```suggestion
Task attributes can be validated using Active Model Validations. Attributes are
validated before a Task is enqueued.

If an attribute uses an inclusion validator with a supported `in:` option, the
set of values will be used to populate a dropdown in the user interface. The
following types are supported:
```
